modules/gov/handle_msg.go

```diff
@@ -32,37 +30,35 @@ func (m *Module) HandleMsg(index int, msg sdk.Msg, tx *juno.Tx) error {
 
 	switch cosmosMsg := msg.(type) {
 	case *govtypesv1.MsgSubmitProposal:
-		return m.handleMsgSubmitProposal(tx, index, cosmosMsg)
+		return m.handleSubmitProposalEvent(tx, cosmosMsg.Proposer, tx.Logs[index].Events)
+	case *govtypesv1beta1.MsgSubmitProposal:
+		return m.handleSubmitProposalEvent(tx, cosmosMsg.Proposer, tx.Logs[index].Events)
 
 	case *govtypesv1.MsgDeposit:
-		return m.handleMsgDeposit(tx, cosmosMsg)
+		return m.handleDepositEvent(tx, cosmosMsg.Depositor, tx.Logs[index].Events)
+	case *govtypesv1beta1.MsgDeposit:
+		return m.handleDepositEvent(tx, cosmosMsg.Depositor, tx.Logs[index].Events)
 
 	case *govtypesv1.MsgVote:
-		return m.handleMsgVote(tx, cosmosMsg)
+		return m.handleVoteEvent(tx, cosmosMsg.Voter, tx.Logs[index].Events)
+	case *govtypesv1beta1.MsgVote:
+		return m.handleVoteEvent(tx, cosmosMsg.Voter, tx.Logs[index].Events)
 
 	case *govtypesv1.MsgVoteWeighted:
-		return m.handleMsgVoteWeighted(tx, cosmosMsg)
+		return m.handleVoteEvent(tx, cosmosMsg.Voter, tx.Logs[index].Events)
+	case *govtypesv1beta1.MsgVoteWeighted:
+		return m.handleVoteEvent(tx, cosmosMsg.Voter, tx.Logs[index].Events)
 	}
 
 	return nil
 }
 
-// handleMsgSubmitProposal allows to properly handle a MsgSubmitProposal
-func (m *Module) handleMsgSubmitProposal(tx *juno.Tx, index int, msg *govtypesv1.MsgSubmitProposal) error {
+// handleSubmitProposalEvent allows to properly handle a handleSubmitProposalEvent
+func (m *Module) handleSubmitProposalEvent(tx *juno.Tx, proposer string, events sdk.StringEvents) error {
```

Comment:
V1beta1 and V1 messages are sharing the same event, so we can parse event rather than push message directly so that we don't support many version messages later.

https://docs.cosmos.network/main/build/modules/gov#events